students/jesse_miller/session09/cli_main.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python3
+'''
+This is the command interface for the mailroom.  Now that I know that donor
+manipulations work correctly (At least according to pytest.  I mean, I wrote
+the tests so the likelihood of human error here is high, you know?).
+'''
+#import os
+import sys
+from donor_models import DonorCollection, DonorTools
+from mail_box import MailBox
+'''
+Module imports
+'''
+
+#pylint: disable=C0103
+name = ''
+alms = DonorCollection()
+tools = DonorTools(name)
+'''
+Because my sense of humor is odd, that's why.  Also setting an object
+'''
+
+class MailRoom:
+    '''
+    All the calls and commands will be through this class.  Manipulation of
+    donor information is done in donor_models.py, and all of the execution for
+    this subprogram is from mailroom_oo.py
+    '''
+    @staticmethod
+    def list_donors():
+        """
+        Lists existing donors.
+        """
+        print(f"\n{'-'*15}\nList of Donors:\n{'-'*15}")
+        current = alms.list_donor()
+        for donor in current:
+            print(donor)
+        print(f"{'-'*15}\n")
+
+
+    @staticmethod
+    def print_report():
+        """
+        Prints a summary report of all donors.
+        """
+        headers = ["Donor Name", "Total Given", "Times Donated", "Average Gift"]
+        print(f"\n{'-'*80}\n{{:17}} | {{:<19}} | {{:<15}} | {{:<19}}\n{'-'*80}"\
+        .format(headers[0], headers[1], headers[2], headers[3]))
+
+        donor_data = alms.create_report()
+        for row in donor_data:
+            print('{:17} | ${:<18,.2f} | {:<15} | ${:<16,.2f}'.format\
+            (row[0], row[1], row[2], row[3]))
+        print(f"{'-'*80}\n")
+
```

Comment:
Try not to use _create_report_ and instead generate the report directly here by pulling _alms.donors_dict_:

```
    @staticmethod
    def print_report():
        """
        Prints a summary report of all donors.
        """
        headers = ["Donor Name", "Total Given", "Times Donated", "Average Gift"]
        print(f"\n{'-'*80}\n{{:17}} | {{:<19}} | {{:<15}} | {{:<19}}\n{'-'*80}"\
        .format(headers[0], headers[1], headers[2], headers[3]))

        donor_data = alms.donors_dict
        for donor in donor_data:
             print('{:17} | ${:<18,.2f} | {:<15} | ${:<16,.2f}'.format\
            (donor_data[donor].name, sum(donor_data[donor].donations), len(donor_data[donor].donations), sum(donor_data[donor].donations)/len(donor_data[donor].donations)))           
        print(f"{'-'*80}\n")
```

Reply:
That makes more sense.  Thanks!